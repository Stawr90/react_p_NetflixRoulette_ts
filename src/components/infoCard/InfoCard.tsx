import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

import { movieCard } from '../moviesList/moviesSlice';

import './infoCard.scss';

const InfoCard = () => {
    const card = useSelector(movieCard);
    const {img, title, date, timer, descr} = card;

    return (
        <div className='info' data-testid="info-card-page">
            <div className="info__img">
                <img src={img} alt={title}/>
            </div>
            <div className="info__text">
                <div className="info__title">{title}</div>
                <div className="info__char">
                    <span className='info__char-year'>{date}</span>
                    <span className='info__char-time'>{timer} min</span>
                </div>
                <div className="info__descr">{descr}</div>
                <Link to='/' data-testid="main-link"><button className="info__btn">Films</button></Link>
            </div>
        </div>
    )
}

export default InfoCard;