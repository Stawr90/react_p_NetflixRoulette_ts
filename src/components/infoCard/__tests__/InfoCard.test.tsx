import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import * as reduxHooks from 'react-redux';

import InfoCard from '../InfoCard';

const infoCard = {
    "img": "https://avatars.mds.yandex.net/get-kinopoisk-image/1599028/2f48ce0d-8590-4ac9-b130-8073f34f5c83/300x450",
    "title": "One of us",
    "date": 2017,
    "timer": 119,
    "descr": "The film follows the lives of three ex-members of Brooklyn is Hasidic community: Ari Hershkowitz, Luzer Twersky, and Etty Ausch. Each struggles with being ostracized from their former community and families, while revealing how they came to leave."
};

const MockInfoCard = () => {
    return (
        <MemoryRouter>
            <InfoCard/>
        </MemoryRouter>
    )
}

jest.mock('react-redux');
const mockedUseSelector = jest.spyOn(reduxHooks, 'useSelector');

describe('InfoCard Component', () => {
    test('must accept correct data', async () => {
        mockedUseSelector.mockReturnValue(infoCard)
        const component = render(<MockInfoCard/>);
        expect(component).toMatchSnapshot();
    });

    test('should render info card', () => {
        render(<MockInfoCard/>);

        expect(screen.getByRole('img')).toBeInTheDocument();
        expect(screen.getByAltText(/one of us/i)).toBeInTheDocument();
        expect(screen.getByText(/one of us/i)).toBeInTheDocument();
        expect(screen.getByText(/2017/i)).toBeInTheDocument();
        expect(screen.getByText(/119 min/i)).toBeInTheDocument();
        expect(screen.getByText(/the film follows/i)).toBeInTheDocument();
        expect(screen.getByRole('button', {name: 'Films'})).toBeInTheDocument();
    });
});