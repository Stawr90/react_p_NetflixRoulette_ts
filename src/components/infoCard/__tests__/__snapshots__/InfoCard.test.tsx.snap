// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard Component must accept correct data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="info"
        data-testid="info-card-page"
      >
        <div
          class="info__img"
        >
          <img
            alt="One of us"
            src="https://avatars.mds.yandex.net/get-kinopoisk-image/1599028/2f48ce0d-8590-4ac9-b130-8073f34f5c83/300x450"
          />
        </div>
        <div
          class="info__text"
        >
          <div
            class="info__title"
          >
            One of us
          </div>
          <div
            class="info__char"
          >
            <span
              class="info__char-year"
            >
              2017
            </span>
            <span
              class="info__char-time"
            >
              119
               min
            </span>
          </div>
          <div
            class="info__descr"
          >
            The film follows the lives of three ex-members of Brooklyn is Hasidic community: Ari Hershkowitz, Luzer Twersky, and Etty Ausch. Each struggles with being ostracized from their former community and families, while revealing how they came to leave.
          </div>
          <a
            data-testid="main-link"
            href="/"
          >
            <button
              class="info__btn"
            >
              Films
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="info"
      data-testid="info-card-page"
    >
      <div
        class="info__img"
      >
        <img
          alt="One of us"
          src="https://avatars.mds.yandex.net/get-kinopoisk-image/1599028/2f48ce0d-8590-4ac9-b130-8073f34f5c83/300x450"
        />
      </div>
      <div
        class="info__text"
      >
        <div
          class="info__title"
        >
          One of us
        </div>
        <div
          class="info__char"
        >
          <span
            class="info__char-year"
          >
            2017
          </span>
          <span
            class="info__char-time"
          >
            119
             min
          </span>
        </div>
        <div
          class="info__descr"
        >
          The film follows the lives of three ex-members of Brooklyn is Hasidic community: Ari Hershkowitz, Luzer Twersky, and Etty Ausch. Each struggles with being ostracized from their former community and families, while revealing how they came to leave.
        </div>
        <a
          data-testid="main-link"
          href="/"
        >
          <button
            class="info__btn"
          >
            Films
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
